java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/ProfileActivity.java

```diff
@@ -34,11 +34,59 @@
 import android.widget.AdapterView.OnItemSelectedListener;
 
 public class ProfileActivity extends Activity {
+	private HealthVaultApp mService;
+	private HealthVaultClient mHVClient;
+	private Record mCurrentRecord;
+
 	@Override
 	protected void onCreate(@Nullable Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 		setContentView(R.layout.profile);
+		mService = HealthVaultApp.getInstance();
+		mHVClient = new HealthVaultClient();
 
 		setTitle("Profile Sample");
 	}
+
+	private void populateProfile() {
+		final EditText firstNameEditText = (EditText) findViewById(R.id.firstNameText);
+		final EditText secondtNameEditText = (EditText) findViewById(R.id.lastNameText);
+
+		String name = mCurrentRecord.getName();
+		String[] firstLastName = name.split(" ");
```

Comment:
If the user has only 1 name, or no space in their name then the split will return an array with 1 item. In this case we will have a NPE below when we do firstLastName[1]. Add some bounds checking.